### PR TITLE
Adding KSonnet configuration of Screwdriver

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This is _very_ specific towards running in AWS right now.  It depends on a number of AWS specific features:

 - Elastic Load Balancers with IP restrictions and HTTPS support via ACM
 - Role delegation via Kube2IAM for writing to S3

### CODE REVIEW

File: environments/default/params.libsonnet [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        password: "INSERT REAL VALUES HERE",
+      },
+      // Used for encrypting session data
+      cookiePassword: "INSERT REAL VALUES HERE",
+      // Used for encrypting stored secrets
+      vaultPassword: "INSERT REAL VALUES HERE",
+    },
+    store +: {
+      // Amazon IAM role for the Store to interact with S3
+      role: "arn:aws:iam::1234567890:role/screwdriver-role-12345",
```
Comment (jithine) [inline environments/default/params.libsonnet:50]: should we leave these as default ?

File: components/config/api.libsonnet [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+          oauthClientSecret: clientSecret,
+          secret: secret,
+          privateRepo: privateRepo
+        },
+      },
+    },
+  },
+
+  executors:: {
+    k8s(namespace="default", version="stable", resources=null):: {
```
Comment (jithine) [inline components/config/api.libsonnet:78]: we should enhance this with rest of executors. For a first pass this is fantastic.

**Review by jithine — APPROVED** [review]
Couple minor comments otherwise LGTM. Took a bit to learn weird jsonnet.

File: environments/default/params.libsonnet (hunk 1 above)
Comment (tkyi) [reply environments/default/params.libsonnet:50]: looks like a dummy value so it should be okay

**Review by tkyi — APPROVED** [review]
Looks good for a first pass. Really cool! 👍

**Pull request merged**